custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go

```diff
@@ -201,6 +203,30 @@ func (p *StackdriverProvider) ListAllMetrics() []provider.MetricInfo {
 	return p.translator.GetMetricsFromSDDescriptorsResp(response)
 }
 
+// GetExternalMetric queries Stackdriver for external metrics.
+func (p *StackdriverProvider) GetExternalMetric(namespace string, metricNameEscaped string, metricSelector labels.Selector) (*external_metrics.ExternalMetricValueList, error) {
+	metricName := unescapeMetricName(metricNameEscaped)
+	stackdriverRequest, err := p.translator.GetExternalMetricRequest(metricName, metricSelector)
+	if err != nil {
+		return nil, err
+	}
+	stackdriverResponse, err := stackdriverRequest.Do()
+	if err != nil {
+		return nil, err
+	}
+	externalMetricItems, err := p.translator.GetRespForExternalMetric(stackdriverResponse, metricNameEscaped)
+	if err != nil {
+		return nil, err
+	}
+	return &external_metrics.ExternalMetricValueList{
+		Items: externalMetricItems,
+	}, nil
+}
+
+func unescapeMetricName(metricName string) string {
+	return strings.Replace(metricName, "|\\", "/", -1)
```

Comment:
Can we change this to just a pipe ('|')? I wanted a longer escape sequence when it was meant for multiple different systems, but AFAIK pipe is not allowed in sd names, so it's good enough by itself.

Reply:
Done